contracts/bonding/BondingManager.sol

```diff
@@ -226,20 +230,25 @@ contract BondingManager is ManagerProxyTarget, IBondingManager {
 
             // Check if transcoder is not already registered
             if (transcoderStatus(msg.sender) == TranscoderStatus.NotRegistered) {
-                if (transcoderPool.isFull()) {
+                if (!transcoderPool.isFull()) {
+                    // If pool is not full add new transcoder
+                    transcoderPool.insert(msg.sender, delegatedAmount, address(0), address(0));
+                } else {
                     address lastTranscoder = transcoderPool.getLast();
-                    // Must have more stake than the last transcoder in the pool
-                    require(delegatedAmount > transcoderPool.getKey(lastTranscoder));
 
-                    transcoderPool.remove(lastTranscoder);
+                    if (delegatedAmount > transcoderPool.getKey(lastTranscoder)) {
```

Comment:
Should this be `delegatedAmount` or `delegatedAmount + t.<current-bonded-stake>`? Because they may already have some amount delegated and this new delegatedAmount could push them over the edge?

Reply:
Ah, nevermind this. There is no additional amount bonded in this function so we don't need to do additional accounting.